### DISCUSSION TITLE
Any ever intersected 3D volumes in python?

### DISCUSSION THREAD

**Discussion by RyanConway91** [modflowpy/flopy#1443] [Q&A] [answered]
I am working on model with an unstructured grid (quadtree, different grid in each layer) and I want to intersect 3D volumes from a mine plan with the grid to figure out what cells are ore body, open shafts, ect. 
 
Anyone ever do this? If so, what was your approach?

**Comment by RyanConway91** [comment, marked as answer]
After some research, I found a way to intersect volumes with paraview using the resample_with_data filter.

**Reply by vcantarella** [reply]
Hi Ryan, Thanks for leaving an answer. I wonder if you could detail your workflow a bit more. How did you export your model/grid to paraview? and did you assign a variable or something and export it back to flopy? Thanks

**Reply by RyanConway91** [reply]
General process is:
1. Export vtk with flopy, adding a fake "unique node" property. Note I had to install the development candidate for the vtk export to work correctly :
```
import flopy
import numpy as np
sim_ws = r"G:\EGC\Talon_Test\Model_Files"
sim = flopy.mf6.MFSimulation.load(sim_name="mfsim.nam", version='mf6',sim_ws= sim_ws, 
                                  load_only=['NPF'], verbosity_level=0)

model = sim.get_model("mf6model")
#reset k33 to be a uni node number. Probably a smarter way to get this...
k = model.npf.k.array
uni_nodes = np.zeros_like(k)+range(0,len(k))
model.npf.k33.set_data(uni_nodes)
outfolder = r"G:\RJC3\Projects\talon_groundwater_modeling"
model.npf.export(outfolder,fmt = "vtk",name = "npf")

```
2. Load the grid vtk into paraview
3. Load the volume you want to intersect into paraview
4. Use resample_with_data filter to "map" the volume back to grid. Check "Mark blank points and cell" and "Pass cell arrays"
5. Use "threshold" filter remove the blank cells (ghostnodes==0)
6. You now have the grid cells that intersect the volume. You export this is a csv and use the unique node number to select grid cells as needed

This is what I got out:
![image](https://user-images.githubusercontent.com/25184697/179282661-344a9c62-9bd1-4a96-be68-4c6bc0bbd694.png)

**Reply by vcantarella** [reply]
Thanks a lot!! Very ingenious!! I will try that out.
